Mathlib/RingTheory/Localization/Module.lean

```diff
@@ -143,6 +143,17 @@ theorem Basis.localizationLocalization_repr_algebraMap {ι : Type _} (b : Basis
     _ = algebraMap R Rₛ (b.repr x i) := by simp [Algebra.smul_def]
 #align basis.localization_localization_repr_algebra_map Basis.localizationLocalization_repr_algebraMap
 
+theorem Basis.localizationLocalization_span {ι : Type _} (b : Basis ι R A) :
+    Submodule.span R (Set.range (b.localizationLocalization Rₛ S Aₛ)) =
+      Submodule.map (IsScalarTower.toAlgHom R A Aₛ) ⊤ := by
+  convert congrArg (Submodule.map (IsScalarTower.toAlgHom R A Aₛ)) b.span_eq
+  change _ = map (AlgHom.toLinearMap _) _
```

Comment:
Can you try to use
```
{F : Type _} [SemilinearMapClass F σ₁₂ M M₂]
```
and `(f : F)` in `Submodule.map_span`? This should make the `change` line useless. I see that most of `Mathlib.LinearAlgebra.Span` already uses `SemilinearMapClass`, it should be a pretty easy generalization.

Reply:
I am sorry, but I cannot see what you want me to do. I tried to remove the `change` and provide some explicit arguments to `Submodule.map_span` but I could not get anywhere without making things more complicated

Reply:
I've pushed a commit, let's see what CI think thinks. If it becomes too complicated feel free to revert it.

Reply:
The point is that `Submodule.map_span` is stated for `f : M →ₛₗ[σ₁₂] M₂` but it actually holds for any `(f : F)`, where `[SemilinearMapClass F σ₁₂ M M₂]`. You have `f : A →ₐ[R] Aₛ`, that is not literally a semilinear map, so you need the `change` line. But we have `[SemilinearMapClass (A →ₐ[R] Aₛ) ... ...]`, so with this generalization the `change` line should become useless.

In general, when a `change` is needed I see this as a warning that a lemma is missing (in this case we can prove, by `rfl`, that `Submodule.map f M = Submodule.map f.toLinearMap M` and rewrite this) or that something is stated in a suboptimal way.

Reply:
Ok. I see. Thanks for the explanation!

Reply:
The reason why rewriting the `rfl` lemma is better than using `change` is the following: suppose that the `change` trick is used 100 times and I refactor the definition of `Submodule.map` in such a way that the `Submodule.map f M = Submodule.map f.toLinearMap M` is not true definitionally anymore (this seems unlike, I agree, but we changed the definition of groups two years ago...). Then I need to fix all the 100 occurrences of the trick, by hand. If instead `rw` is used, I only need to fix it once, in the proof of the lemma.